letsencrypt/models/base_config_settings.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class BaseConfigSettings(models.TransientModel):
+    _inherit = 'base.config.settings'
+
+    letsencrypt_dns_provider = fields.Selection([], string='DNS Provider')
+    letsencrypt_login = fields.Char(string='Login')
+    letsencrypt_key = fields.Text(string='Key')
+    letsencrypt_url = fields.Char(string='URL')
+    letsencrypt_needs_dns_provider = fields.Boolean(
```

Comment:
shouldn't this be a computed field that looks at the configured domains and decides if we need DNS validation?